### PR TITLE
feat: implement InputMediaBuilder

### LINKED ISSUE #101: feat: Media group builder
Just like we have a builder for `InlineKeyboard` objects, we want one that simplifies creating media group objects which can be passed to the `media` property of https://core.telegram.org/bots/api#sendmediagroup

### PR DESCRIPTION
**Pull request by WingLim**
**Create `builders` subdirectory which other builders can put themselves in it.**

InputMediaBuilder methods:

Basic methods:

- add
- build

Advanced methods:

All these methods use `Omit` to remove `type` property for each different `InputMedia` type, so that user can use it more convenient.

- photo
- video
- animation
- audio
- document

Static method:

`from` method to create a InputMediaBuilder with `InputMedia`.

- from

Furthermore, it have `constructor` to initialize with some `InputMedia`.

It means if you like OOP, you can use `new  InputMediaBuilder(/** data*/)`, or you can use `InputMediaBuilder.from(/** data*/)` if you like FP more.

TODO:
- [ ] Documentation for InputMediaBuilder

Close #101

### CODE REVIEW

**Comment by WingLim** [comment]
@KnorpelSenf Plz take a look the code, if it's good for you, then I will write the documentation for it.

**Review by roj1512 — COMMENTED** [review]
I recommend changing the file name to something simpler (one word), or to `input_media_builder.ts`. So it is consistent with either the other grammY files or official Deno libraries.

**Comment by WingLim** [comment]
The static method `from` comes from #104, and according to https://github.com/grammyjs/grammY/pull/211#issuecomment-1128112737, we can use constructor to do this job.

I don't know which one is better?

**Comment by WingLim** [comment]
> The static method `from` comes from #104, and according to [#211 (comment)](https://github.com/grammyjs/grammY/pull/211#issuecomment-1128112737), we can use constructor to do this job.
> 
> I don't know which one is better?

I have added them both, in case someone like the other style more.

File: src/convenience/builders/input_media.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+     * The nested array that holds the media.
+     */
+    public readonly media: InputMedia[] = [];
+
+    /**
+     * Initialize a new `InputMediaBuilder` with optional `InputMedia` array.
+     *
+     * @param media The media to initialize the builder with
+     */
+    constructor(media?: InputMedia[]) {
```
Comment (KnorpelSenf) [inline src/convenience/builders/input_media.ts:21]: This can be merged with the field declaration.

File: src/convenience/builders/input_media.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
+import {
+    InputMedia,
+    InputMediaAnimation,
+    InputMediaAudio,
+    InputMediaDocument,
+    InputMediaPhoto,
```
Comment (KnorpelSenf) [inline src/convenience/builders/input_media.ts:6]:
```suggestion
    type InputMedia,
    type InputMediaAnimation,
    type InputMediaAudio,
    type InputMediaDocument,
    type InputMediaPhoto,
```

**Review by KnorpelSenf — CHANGES_REQUESTED** [review]
What do you think about also providing something like `InputMediaBuilder.Photo` and the like? That way, we could create stuff using `InputMediaBuilder.Photo.from(...)` which can be passed to `InputMediaBuilder.from` in turn. In addition, it allows for `new InputMediaBuilder.Photo()` with chained method calls. The main builder could then also support the sub-builders, so that `build` does not have to be called on all of them.

I am also wondering if there is a better way to create these things than specifying a config object for every entry. Perhaps this is already possible with the idea described above?

**Comment by WingLim** [comment]
I have a idea about this.

Add a `fromType` static method which accept 2 params:
1. The media array
2. The type of `InputMedia`

Then use a `GenericInputMediaBuilder` to constraint the type of the specific `InputMedia`(e.g. `InputMediaPhoto`) builder.

Let me show you the code to explain it more.

**Comment by KnorpelSenf** [comment]
Please also add some tests.

**Review by KnorpelSenf — CHANGES_REQUESTED** [review]
I now think it is overkill to have a builder for an array. I remember thinking that I would like to abstract that away, but now I believe this would be overkill and take it too far.

I used the rest your changes to create #437. This will subsequently be closed. Thank you anywy for this contribution (and sorry for the long delay)!

**Pull request closed without merging**